src/io/file_descriptor.cr

```diff
@@ -149,6 +149,14 @@ class IO::FileDescriptor < IO
     system_fsync(flush_metadata)
   end
 
+  # Write to a specific position in a file
+  # It will flush the current write buffer and not buffer this write.
+  def write_at(bytes : Bytes, offset)
+    check_open
+    flush
+    unbuffered_write_at(bytes, offset)
```

Comment:
Since there's no buffering involved, I'd suggest not to follow the naming scheme for buffered operations. Maybe we could call the method `system_write_at`?